### PR TITLE
USWDS-Sandbox: Zombie defense component updates

### LINKED ISSUE #99: Zombie training prototype changes for August 2023 testing
We are planning to conduct a new round of usability testing (based on issues found in previous testing) for the following components:

- date range picker
- combobox
- input mask
- character count
- tooltip
- validation
- accordion
- file input (new!)

We need to make some changes to some of the component functionality based on what improvements we aim to test in this next round. Some components have no changes to test as we need to gather more data on them first.

**Components that need updates (based on recent improvements we've made) in the prototype:**
- combobox (better navigation using keystrokes and tab/spacebar)
- input mask (include recent improvements from [#5227](https://github.com/uswds/uswds/pull/5227))
- character count (delayed audio feedback, consider adding hint text above the field for screen magnifier users)
- add file input (add most updated version to "Step 1 of 4: Training Details" page)

Everything else stays the same. 

**Changes needed in the [prototype](https://federalist-d5e7c07c-6ffa-4b8e-935f-49a7df24a505.sites.pages.cloud.gov/preview/uswds/uswds-sandbox/jb-zombie-defense-training/) are documented in this [component testing planning mural board](https://app.mural.co/invitation/mural/gsa6/1682975695042?sender=u83d0e400125d69b595f75755&key=4265205b-4557-4ce8-99fe-219d854729c2).** Go to the "Prototype: Zombie defense Training Session" section. Changes are documented with yellow sticky notes.

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

Implements changes documented in this [component testing planning mural board](https://app.mural.co/invitation/mural/gsa6/1682975695042?sender=u83d0e400125d69b595f75755&key=4265205b-4557-4ce8-99fe-219d854729c2).

This branch also updates to the latest `uswds-sandbox` repo
> **Note**
> To minimize changed files, we should pull the latest changes into the target branch

## Related Issue
Resolves #99 

## Preview link
[Zombie Defense Training →](https://federalist-d5e7c07c-6ffa-4b8e-935f-49a7df24a505.sites.pages.cloud.gov/preview/uswds/uswds-sandbox/cm-zombie-defense-updates/)

## Changes

- Implements state combobox
   - In the miro page, this was originally proposed to be dependant on the `in-person course` option being selected
   - Dan mentioned **not** relying on a logic decision to display any form elements, but to have them all be readily available.
   - Instead, I just added the state combobox to the default form page
- Implements file upload component
- Implements step indicator component
- Updates input mask pattern
  - New pattern: `AAA-###`
  - Old pattern: `A#A-#A#`
- Installs latest uswds `3.6.0` package
  - Corrects input mask hover style issue

### Exclusions

- Combobox keyboard navigations were improved with https://github.com/uswds/uswds/pull/5160 and included with the USWDS update.
- Input mask improvements in #[5227](https://github.com/uswds/uswds/pull/5227) are still in progress
- Character Count hint text was not adjusted as discussed with @jaclinec
   - Dan mentioned not wanting to move the `25 characters allowed` from below the text input
   - Adding a separate label would be redundant and cause screen readers to announce the character limit twice
   - Most screen readers will be able to provide the information to users, this is only a _potential_ difficulty with screen magnifier users

### CODE REVIEW

**Comment by mahoneycm** [comment]
## Updates after discussion w/ @jaclinec 

- [ ] Input mask has disabled hover state
   - Current USWDS bug. Will be resolved with `3.5.1`
- [x] Updated alphanumeric pattern to `AAA-###`
- [x] Looked into character count allowing input past limit
   - This is an expected pattern and how the component natively acts
- [x] Implemented step indicator

**Comment by mahoneycm** [comment]
## Update 8.25

Installed latest USWDS package which resolved the input mask hover style issue

File: src/instructor-details.html [hunk 1]
```diff
@@ -7,10 +7,56 @@
...
+                    ><span class="usa-sr-only">Step</span>
+                    <span class="usa-step-indicator__current-step">3</span>
+                    <span class="usa-step-indicator__total-steps">of 4</span> </span
+                  ><span class="usa-step-indicator__heading-text"
+                    >Instructor Details</span
+                  >
+                </h1>
+              </div>
+            </div>
+
```
Comment (mejiaj) [inline src/instructor-details.html:59]: Consider moving this to an include to make it easier to update.

**Review by mejiaj — APPROVED** [review]

**Comment by mahoneycm** [comment]
Closing as this work is no longer needed

**Pull request closed without merging**